### PR TITLE
Cms blocks fix

### PR DESCRIPTION
**Pull request by dmitrijs-voronovs**
Added enabled/disabled support for CMS blocks.

Related to `source` PR: https://github.com/scandipwa/base-theme/pull/276

### CODE REVIEW

File: src/Model/Resolver/DataProvider/Block.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+     * @return array
+     * @throws NoSuchEntityException
+     */
+    public function getData(string $blockIdentifier): array
+    {
+        $block = $this->blockRepository->getById($blockIdentifier);
+
+        if (false === $block->isActive()) {
+            return [
+                'disabled' => true
```
Comment (alfredsgenkins) [inline src/Model/Resolver/DataProvider/Block.php]: I believe you need to return the `BlockInterface::IDENTIFIER => $block->getIdentifier(),` as well.

File: src/Model/Resolver/DataProvider/Block.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+     *
+     * @param string $blockIdentifier
+     * @return array
+     * @throws NoSuchEntityException
+     */
+    public function getData(string $blockIdentifier): array
+    {
+        $block = $this->blockRepository->getById($blockIdentifier);
+
+        if (false === $block->isActive()) {
```
Comment (alfredsgenkins) [inline src/Model/Resolver/DataProvider/Block.php]: Please put expression before Boolean compartment. You can also write: `!$block->isActive()`.

File: src/Model/Resolver/DataProvider/Block.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            return [
+                'disabled' => true
+            ];
+        }
+
+        $renderedContent = $this->widgetFilter->filter($block->getContent());
+
+        $blockData = [
```
Comment (alfredsgenkins) [inline src/Model/Resolver/DataProvider/Block.php]: Why to declare `$blockData`, you can return `[ ... ]` immediately.

File: src/Model/Resolver/DataProvider/Block.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+     *
+     * @param string $blockIdentifier
+     * @return array
+     * @throws NoSuchEntityException
+     */
+    public function getData(string $blockIdentifier): array
+    {
+        $block = $this->blockRepository->getById($blockIdentifier);
+
+        if (!$block->isActive()) {
```
Comment (eli-l) [inline src/Model/Resolver/DataProvider/Block.php]: Why do we need to return non-active blocks?

File: src/etc/schema.graphqls [hunk 5]
```diff
@@ -3,4 +3,11 @@ type Query {
         id: Int @doc(description: "Id of the CMS page")
         url_key: String @doc(description: "Url key of the CMS page")
     ): CmsPage @resolver(class: "Magento\\CmsGraphQl\\Model\\Resolver\\Page") @doc(description: "The CMS page query returns information about a CMS page")
-}
\ No newline at end of file
+}
+
+type CmsBlock @doc(description: "CMS block defines all CMS block information") {
```
Comment (eli-l) [inline src/etc/schema.graphqls]: Please do not re-define already defined fields

File: src/Model/Resolver/DataProvider/Block.php (hunk 4 above)
Comment (dmitrijs-voronovs) [reply src/Model/Resolver/DataProvider/Block.php]: The task was `Make it so graphqlQL does not throw error, but return disabled or similar.`
Therefore it was implemented like this.

Comment (dmitrijs-voronovs) [reply src/Model/Resolver/DataProvider/Block.php]: Should I implement it otherwise?

**Pull request merged**